examples/azure/terraform-azure-availabilityset-example/main.tf

```diff
@@ -0,0 +1,104 @@
+# DEPLOY AVAILABILITY SET
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_availability_set" "main" {
+  name                        = "${var.prefix}-example"
+  location                    = azurerm_resource_group.main.location
+  resource_group_name         = azurerm_resource_group.main.name
+  platform_fault_domain_count = var.avs_fault_domain_count
+  managed                     = true
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY VIRTUAL NETWORK RESOURCES
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_virtual_network" "main" {
+  name                = "${var.prefix}-network"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+}
+
+resource "azurerm_subnet" "internal" {
+  name                 = "internal"
+  resource_group_name  = azurerm_resource_group.main.name
+  virtual_network_name = azurerm_virtual_network.main.name
+  address_prefixes       = ["10.0.17.0/24"]
+}
+
+resource "azurerm_network_interface" "main" {
+  name                = "${var.prefix}-nic"
+  location            = azurerm_resource_group.main.location
+  resource_group_name = azurerm_resource_group.main.name
+
+  ip_configuration {
+    name                          = "terratestconfiguration1"
+    subnet_id                     = azurerm_subnet.internal.id
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY VIRTUAL MACHINE TO THE AVAILABILITY SET
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_virtual_machine" "main01" {
+  name                             = "${var.prefix}-vm"
+  location                         = azurerm_resource_group.main.location
+  resource_group_name              = azurerm_resource_group.main.name
+  network_interface_ids            = [azurerm_network_interface.main.id]
+  availability_set_id              = azurerm_availability_set.main.id
+  vm_size                          = var.vm_size
+  delete_os_disk_on_termination    = true
+  delete_data_disks_on_termination = true
+
+  storage_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "18.04-LTS"
+    version   = "latest"
+  }
+
+  storage_os_disk {
+    name              = "${var.prefix}-osdisk"
+    caching           = "ReadWrite"
+    create_option     = "FromImage"
+    managed_disk_type = "Standard_LRS"
+  }
+
+  os_profile {
+    computer_name  = "${var.prefix}-vm"
+    admin_username = var.username
+    admin_password = var.password
```

Comment:
What's the password for? Is this an SSH password? This brings up all sorts of security implications, so I'm wondering if we really need it for this example?

Reply:
@davesee, can you look at dropping the os_profile from the example?  https://www.terraform.io/docs/providers/azurerm/r/virtual_machine.html#os_profile.  Not sure if that affects the test or not.

Reply:
It is necessary for the VM creation unfortunately. I put a note in the variable.tf file to highlight the implications.

variable.tf > password > description:
"Password for the VM OS, for demonstration ONLY! This should not be used with production resources"

Test error without OS Terraform section:
Error: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidParameter" Message="Required parameter 'osProfile' is missing (null)." Target="osProfile"
...
--- FAIL: TestTerraformAzureAvailabilitySetExample (287.60s)

Reply:
Since the VM is not being accessed, I can use the Terraform random_password generator to ensure stagnant resources are not compromised.

https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/password